pkg/util/cache/cache.go

```diff
@@ -349,6 +349,11 @@ func (c *lruCache[K, V]) getAndPin(ctx context.Context, key K) (*cacheItem[K, V]
 		timer := time.Now()
 		value, err := c.loader(ctx, key)
 		c.stats.TotalLoadTimeMs.Add(uint64(time.Since(timer).Milliseconds()))
+		// Try to evict one item if there is not enough disk space, then retry.
+		if merr.ErrServiceDiskLimitExceeded.Is(err) {
+			c.evictItems(ctx, 1)
+			value, err = c.loader(ctx, key)
```

Comment:
It's a just a retry, does not promise to release enough disk space to do loading.
And at huge concurrent request, it may evict many segments from diskcache at one time.

Reply:
> It's a just a retry, does not promise to release enough disk space to do loading. And at huge concurrent request, it may evict many segments from diskcache at one time.

yes, it does not guarantee this eviction will release enough disk space. It's hard to decide how many items we should evict at once and how many times we should retry to load. I can change the eviction number configurable.